go/apps/http_api_nostream/resource.py

```diff
@@ -247,6 +247,13 @@ def handle_PUT_in_reply_to(self, request, payload, in_reply_to):
     @inlineCallbacks
     def handle_PUT_send_to(self, request, payload):
         user_account = request.getUser()
+        optout_store = self.worker.vumi_api.get_user_api(user_account) \
+            .optout_store
+        optout = yield optout_store.get_opt_out(
+            payload.get('to_addr_type', 'msisdn'), payload.get('to_addr'))
+        if optout:
+            self.client_error_response(request, "Recipient has opted out")
```

Comment:
I think adding the address and address type to the message would be useful.

I'm not so sure that it shouldn't be an error though, the message that the user wanted to send isn't going to be sent. I think that BAD REQUEST is a good response to this, unless you have suggestions for another response code?